.github/workflows/dapr.yml

```diff
@@ -686,3 +686,13 @@ jobs:
             echo "::error::There is no change for ${daprVersion} Helm chart. Did you forget to update the chart version before tagging?"
             exit -1
           fi
+  rerun:
+    name: Reruns in case of failure
+    needs: helmpublish
+    if: failure() && fromJSON(github.run_attempt) < 3 && github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - env:
+          GH_REPO: ${{ github.repository }}
+          GH_TOKEN: ${{ github.token }}
+        run: gh workflow run rerun.yml -F run_id=${{ github.run_id }}
```

Comment:
Rather than doing this, which IMHO adds complexity and it's hard to follow, let's just add autmatic retries to tests when not building a PR (just one CLI flag passed to gotestsum). Although I think this should only be done for releases.